src/panels/lovelace/cards/hui-picture-card.ts

```diff
@@ -70,6 +80,35 @@ export class HuiPictureCard extends LitElement implements LovelaceCard {
     return false;
   }
 
+  protected willUpdate(changedProps: PropertyValues) {
+    super.willUpdate(changedProps);
+
+    if (!this._config || !this.hass) {
+      return;
+    }
+
+    const firstHass =
+      changedProps.has("hass") && changedProps.get("hass") === undefined;
+    const imageChanged =
+      changedProps.has("_config") &&
+      changedProps.get("_config")?.image !== this._config?.image;
+
+    if (imageChanged) {
+      this._resolvedImage = this._config?.image;
+    }
+
+    if (
+      (firstHass || imageChanged) &&
+      typeof this._config?.image === "string" &&
+      isMediaSourceContentId(this._config.image)
+    ) {
+      this._resolvedImage = null;
+      resolveMediaSource(this.hass, this._config?.image).then((result) => {
+        this._resolvedImage = result.url;
+      });
+    }
```

Comment:
This block can be refactored to only assign `this._resolvedImage` once + the async one, but not a big deal